src/linse/profile.py

```diff
@@ -0,0 +1,55 @@
+from collections import defaultdict
+from linse.segment import ipa
+from linse.models import *
+
+__all__ = ['simple', 'context', 'structured']
+
+class Form(object):
+
+    def __init__(self, text, **kw):
+        self.text = text
+        kw['text'] = text
+        self.kw = kw
+
+    def __str__(self):
+        return self.text
+
+
+def get_profile(
+        *forms, 
+        preceding=None,
+        following=None,
+        segmenter=None,
+        **kw
+        ):
+    """
+    Retrieve a list of graphemes from a list of forms.
+    """
+    if not segmenter:
+        keywords = dict(
+                semi_diacritics='shʃʂɕɦʐʑʒw',
+                merge_vowels=True,
+                merge_geminates=True)
+        keywords.update(kw)
+        segmentize = lambda x: ipa(x['text'], **keywords)
+    else:
+        segmentize = lambda x: segmenter(x['text'], **kw)
+            
+    preceding, following = preceding or '', following or ''
+    graphemes = defaultdict(list)
+    for i, form in enumerate(forms):
+        meta = form.kw if hasattr(form, 'kw') else {'ID': i+1, 'text': form}
```

Comment:
This may seem a bit "umständlich", but I do not want to cast the data to string, since this can yield unwanted segmentations, like "None". Don't know if there is a better solution for this.